reconcilers/child.go

```diff
@@ -306,16 +306,21 @@ func (r *ChildReconciler[T, CT, CLT]) reconcile(ctx context.Context, resource T)
 	c := RetrieveConfigOrDie(ctx)
 
 	actual := r.ChildType.DeepCopyObject().(CT)
-	children := r.ChildListType.DeepCopyObject().(CLT)
-	if err := c.List(ctx, children, r.listOptions(ctx, resource)...); err != nil {
-		return nilCT, err
+	children := RetrieveKnownChildren[CT](ctx)
+	if children == nil {
+		// use existing known children when available, fall back to lookup
```

Comment:
Curious to know what situation can lead to this?

Reply:
It’s the normal path for direct ChildReconciler use. The shortcut is only used by ChildSetReconciler today.

Reply:
Thanks